test/jdk/javax/swing/plaf/basic/BasicTreeUI/TreeCellRendererLeakTest.java

```diff
@@ -0,0 +1,206 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6507038
+ * @summary Verifies memory leak in BasicTreeUI TreeCellRenderer
+ * @run main TreeCellRendererLeakTest
+ */
+
+import java.awt.Component;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTabbedPane;
+import javax.swing.JTree;
+import javax.swing.SwingUtilities;
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.DefaultTreeCellRenderer;
+import javax.swing.tree.DefaultTreeModel;
+import javax.swing.tree.TreeNode;
+
+public class TreeCellRendererLeakTest {
+
+    static long smCount = 0;
+    static boolean done;
+
+    private static JFrame frame;
+    private JPanel jPanel1;
+    private JPanel jPanel2;
+    private JScrollPane jScrollPane1;
+    private JTabbedPane jTabbedPane1;
+    private JTree jTree1;
+
+    // JLabel with an instance counter
+    public class TestLabel extends JLabel {
+        public TestLabel() {
+            smCount++;
+        }
+
+        public void finalize( ) {
+            smCount--;
+        }
+    }
+
+    // Custom TreeCellRenderer
+    public class TreeCellRenderer extends DefaultTreeCellRenderer {
+
+        public TreeCellRenderer( ) {
+        }
+
+        // Create a new JLabel every time
+        public Component getTreeCellRendererComponent(
+                JTree tree,
+                Object value,
+                boolean sel,
+                boolean expanded,
+                boolean leaf,
+                int row,
+                boolean hasFocus) {
+            JLabel label = new TestLabel( );
+            label.setText("TreeNode: " + value.toString());
+            if (sel) {
+                label.setBackground(getBackgroundSelectionColor());
+            } else {
+                label.setBackground(getBackgroundNonSelectionColor());
+            }
+
+            return label;
+        }
+    }
+
+    public TreeCellRendererLeakTest() {
+        initComponents();
+        jTree1.setCellRenderer(new TreeCellRenderer());
+        Thread updateThread = new Thread(new Runnable( ) {
+            public void run( ) {
+                runChanges( );
+            }
+        });
```

Comment:
This could be a method reference:
```suggestion
        Thread updateThread = new Thread(this::runChanges);
```